diff --git a/python/cppyy/__init__.py b/python/cppyy/__init__.py
index 991fee8..961ea5d 100644
--- a/python/cppyy/__init__.py
+++ b/python/cppyy/__init__.py
@@ -70,6 +70,34 @@ if ispypy:
 else:
     from ._cpython_cppyy import *
 
+#- add to the dynamic path as needed -----------------------------------------
+def add_default_paths():
+    gSystem = gbl.gSystem
+    if os.getenv('CONDA_PREFIX'):
+      # MacOS, Linux
+        lib_path = os.path.join(os.getenv('CONDA_PREFIX'), 'lib')
+        if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
+
+      # Windows
+        lib_path = os.path.join(os.getenv('CONDA_PREFIX'), 'Library', 'lib')
+        if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
+
+    if not is_pypy:
+        # assuming that we are in PREFIX/lib/python/site-packages/cppyy, add PREFIX/lib to the search path
+        lib_path = os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir, os.path.pardir, os.path.pardir))
+        if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
+
+    try:
+        with open('/etc/ld.so.conf') as ldconf:
+            for line in ldconf:
+                f = line.strip()
+                if (os.path.exists(f)):
+                    gSystem.AddDynamicPath(f)
+    except IOError:
+        pass
+add_default_paths()
+del add_default_paths
+
 
 #- allow importing from gbl --------------------------------------------------
 sys.modules['cppyy.gbl'] = gbl
diff --git a/python/cppyy/_cpython_cppyy.py b/python/cppyy/_cpython_cppyy.py
index b5939fa..603c69d 100644
--- a/python/cppyy/_cpython_cppyy.py
+++ b/python/cppyy/_cpython_cppyy.py
@@ -96,36 +96,6 @@ gbl.std =  _backend.CreateScopeProxy('std')
 # for move, we want our "pythonized" one, not the C++ template
 gbl.std.move  = _backend.move
 
-
-#- add to the dynamic path as needed -----------------------------------------
-import os
-def add_default_paths():
-    gSystem = gbl.gSystem
-    if os.getenv('CONDA_PREFIX'):
-      # MacOS, Linux
-        lib_path = os.path.join(os.getenv('CONDA_PREFIX'), 'lib')
-        if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
-
-      # Windows
-        lib_path = os.path.join(os.getenv('CONDA_PREFIX'), 'Library', 'lib')
-        if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
-
-  # assuming that we are in PREFIX/lib/python/site-packages/cppyy, add PREFIX/lib to the search path
-    lib_path = os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir, os.path.pardir, os.path.pardir))
-    if os.path.exists(lib_path): gSystem.AddDynamicPath(lib_path)
-
-    try:
-        with open('/etc/ld.so.conf') as ldconf:
-            for line in ldconf:
-                f = line.strip()
-                if (os.path.exists(f)):
-                    gSystem.AddDynamicPath(f)
-    except IOError:
-        pass
-add_default_paths()
-del add_default_paths
-
-
 #- exports -------------------------------------------------------------------
 addressof     = _backend.addressof
 bind_object   = _backend.bind_object
